enterprise/server/util/cacheproxy/cacheproxy.go

```diff
@@ -633,7 +633,7 @@ func (wc *streamWriteCloser) Commit() error {
 	if status.IsAlreadyExistsError(err) {
 		return nil
 	}
-	if sendErr != nil {
+	if err == nil && sendErr != nil {
```

Comment:
This could just be `if err != nil { return err }`, right? (And return `senderr` below)